src/main/java/org/folio/service/orders/CompositePoLineValidationService.java

```diff
@@ -76,6 +76,58 @@ private List<Error> validatePoLineFormats(CompositePoLine compPOL) {
     return Collections.emptyList();
   }
 
+  public List<Error> validatePoLineMaterial(CompositePoLine compPOL) {
+    CompositePoLine.OrderFormat orderFormat = compPOL.getOrderFormat();
+
+    return switch (orderFormat) {
+      case P_E_MIX -> checkPEMix(compPOL);
+      case ELECTRONIC_RESOURCE -> checkElectronicResource(compPOL);
+      case PHYSICAL_RESOURCE -> checkPhysicalResource(compPOL);
+      case OTHER -> checkOtherResource(compPOL);
+      default -> Collections.emptyList();
+    };
+  }
+
+  private List<Error> checkPEMix(CompositePoLine compPOL) {
+    List<ErrorCodes> errors = new ArrayList<>();
+
+    if ((compPOL.getPhysical() == null) || (compPOL.getEresource() == null) || (getElectronicCostQuantity(compPOL) == 0) || (getPhysicalCostQuantity(compPOL) == 0)) {
+      errors.add(ErrorCodes.INVALID_PEMIX_POL);
+    }
+
+    return convertErrorCodesToErrors(compPOL, errors);
+  }
+
+  private List<Error> checkElectronicResource(CompositePoLine compPOL) {
+    List<ErrorCodes> errors = new ArrayList<>();
+
+    if ((compPOL.getPhysical() != null) && (getPhysicalCostQuantity(compPOL) != 0)) {
+      errors.add(ErrorCodes.INVALID_ELECTRONIC_POL);
+    }
+
+    return convertErrorCodesToErrors(compPOL, errors);
+  }
+
+  private List<Error> checkPhysicalResource(CompositePoLine compPOL) {
+    List<ErrorCodes> errors = new ArrayList<>();
+
+    if ((compPOL.getEresource() != null) && (getElectronicCostQuantity(compPOL) != 0)) {
+      errors.add(ErrorCodes.INVALID_PHYSICAL_POL);
+    }
+
+    return convertErrorCodesToErrors(compPOL, errors);
+  }
+
+  private List<Error> checkOtherResource(CompositePoLine compPOL) {
+    List<ErrorCodes> errors = new ArrayList<>();
+
+    if ((compPOL.getEresource() != null) && (getElectronicCostQuantity(compPOL) != 0)) {
+      errors.add(ErrorCodes.INVALID_OTHER_POL);
+    }
+
+    return convertErrorCodesToErrors(compPOL, errors);
+  }
```

Comment:
I checked that we alreade have this validation in the same file:
![image](https://github.com/folio-org/mod-orders/assets/25097693/153afe64-3315-4f90-ae86-5c86c19fa725)

So we just add to add 4th method: validatePoLineWithPEMixFormat
Also we can delete these new error codes and should adjust unit tests. 
I will re-review tomorrow, @yuntianhu does it makes sense?